text/0000-wallet-replay-protection.md

```diff
@@ -0,0 +1,322 @@
+```
+
+This is the main idea used in this proposal to protect against replay attacks. This is why we name the outputs we created along side the `anchor` output `Protected` outputs. They provide protection for subsequent transactions we make if we use them as inputs to our transactions. Key thing to note here is that while we started with 2 protected outputs `O1` and `O2`, the outputs that we create in a transaction to which we add a protected input automatically become `Protected` outputs as well. This property is what allows us to continue creating new protected outputs without the need to create more than one `anchor` output.
+
+
+## Wallet rules for replay protection
+
+Most common transaction types are:
+1. A `Regular` tx which has inputs from a single party - the sender
+2. A `PayJoin` tx which has inputs from both parties - both sender and receiver contribute at least one input each
+
+Let's call outputs that hang from a chain that leads to your anchor a `Protected` output. A transaction is protected from replays if a party contributes a Protected output as an input in the transaction. It's not enough that some other party contributes such an input. We are only really protected from replay attacks if we contribute the `Protected` output. Otherwise, our security depends on the honesty and following of the wallet rules of other participants which should not be relied on.
+
+Let's say that we want to be able to do both `Regular` and `PayJoin` transactions. We do both of these so we have some protected outputs and some that are not protected. The first problem we have is that after a wallet restore, we can't really tell whether an output is protected through a PayJoins+Anchor or not. It would be nice if we could label our outputs as either `Protected` or `Unprotected` after a wallet restore.
+
+### Separation of replay protected outputs
+
+Let's define a clear separation of outputs that are protected from those that are not. We define a protected output generator `GenP` that allows us to either `create` a protected output or `check` whether an output is `Protected`.
+
+An output is labeled as `Protected` if it was created in a transaction where we either contributed a protected input, or contributed an anchor output.
+
+#### Possible protected output generators implementations
+
+As we mentioned above, `GenP` for `Protected` outputs would need to have `create` and `check` functions defined.
+
+There are a few choices how to label outputs as `Protected` while still being able to identify them across difference devices:
+1. Call to `create` uses a new derivation path `P` that is used only for creating `Protected` outputs. Similarly `check` uses the same `P` to check whether an output is protected. Perhaps we could have `N` labels possible which would be labeled by the `r % N` result
+2. Call to `create` creates a specific `r` value for `Protected` outputs e.g. they should start with `N` zeros or be divisible by some number `M`

```

Comment:
I suggest moving this option to a "Rationale & Alternatives" section, since we prefer methods that automatically labels all existing outputs as unprotected.

Reply:
I actually no longer see it relevant tbh, because it has other issues like what if an existing old output matches the constraints etc. I'm going to throw it out because it seems clearly inferior to other approaches.